httpcore/_sync/connection.py

```diff
@@ -54,6 +55,7 @@ def __init__(
         self._connection: Optional[ConnectionInterface] = None
         self._connect_failed: bool = False
         self._request_lock = Lock()
+        self._socket_options = () if socket_options is None else socket_options
```

Comment:
```suggestion
        self._socket_options = socket_options
```

let's keep this optional just now.